src/SDK/SdkAutoloader.php

```diff
@@ -24,7 +24,7 @@ class SdkAutoloader
     public static function autoload(): bool
     {
         try {
-            self::$enabled ??= Configuration::getBoolean(Variables::OTEL_PHP_AUTOLOAD_ENABLED);
+            self::$enabled ??= Configuration::getBoolean(Variables::OTEL_PHP_AUTOLOAD_ENABLED) && !Sdk::isDisabled();
```

Comment:
This should be moved into the initializer callback - currently it's checked during autoloader initialization which makes usage with `.env` files that are loaded afterwards impossible.